crazy_rl/multi_agent/jax/base_parallel_env.py

```diff
@@ -0,0 +1,389 @@
+"""The Base environment inheriting from pettingZoo Parallel environment class."""
+import functools
+
+# import time
+from functools import partial
+from typing import Optional
+from typing_extensions import override
+
+import jax.numpy as jnp
+import jax_dataclasses as jdc
+import numpy as np
+import pygame
+from cflib.crazyflie.swarm import Swarm
+from gymnasium import spaces
+from jax import jit
+from OpenGL.GL import (
+    GL_AMBIENT,
+    GL_AMBIENT_AND_DIFFUSE,
+    GL_BLEND,
+    GL_COLOR_BUFFER_BIT,
+    GL_COLOR_MATERIAL,
+    GL_DEPTH_BUFFER_BIT,
+    GL_DEPTH_TEST,
+    GL_DIFFUSE,
+    GL_FRONT_AND_BACK,
+    GL_LIGHT0,
+    GL_LIGHTING,
+    GL_MODELVIEW,
+    GL_MODELVIEW_MATRIX,
+    GL_ONE_MINUS_SRC_ALPHA,
+    GL_POSITION,
+    GL_PROJECTION,
+    GL_SMOOTH,
+    GL_SRC_ALPHA,
+    glBlendFunc,
+    glClear,
+    glColor4f,
+    glColorMaterial,
+    glEnable,
+    glGetFloatv,
+    glLight,
+    glLightfv,
+    glLineWidth,
+    glLoadIdentity,
+    glMatrixMode,
+    glMultMatrixf,
+    glPopMatrix,
+    glPushMatrix,
+    glShadeModel,
+)
+from OpenGL.raw.GLU import gluLookAt, gluPerspective
+from pettingzoo.utils.env import ParallelEnv
+from pygame import DOUBLEBUF, OPENGL
+
+from crazy_rl.utils.graphic import axes, field, point, target_point
+
+
+# from crazy_rl.utils.utils import run_land, run_sequence, run_take_off
+
+
+@jdc.pytree_dataclass
+class State:
+    """State of the environment containing the modifiable variables."""
+
+    timestep: int
+    agent_location: jnp.ndarray
```

Comment:
Could you order it as in the PZ api for easier reading: obs, reward, terminations, truncations?